src/libraries/System.Drawing.Common/src/System/Drawing/ImageInfo.cs

```diff
@@ -41,27 +41,37 @@ public ImageInfo(Image image)
                     PropertyItem? frameDelayItem = image.GetPropertyItem(PropertyTagFrameDelay);
 
                     // If the image does not have a frame delay, we just return 0.
-                    //
                     if (frameDelayItem != null)
                     {
                         // Convert the frame delay from byte[] to int
-                        //
                         byte[] values = frameDelayItem.Value!;
-                        Debug.Assert(values.Length == 4 * FrameCount, "PropertyItem has invalid value byte array");
-                        _frameDelay = new int[FrameCount];
-                        for (int i = 0; i < FrameCount; ++i)
+
+                        // On Windows, we get the frame delays for every frame. On Linux, we only get the first frame delay.
+                        // We handle this by treating the frame delays as a repeating sequence, asserting that the sequence
+                        // is fully repeatable to match the frame count.
+                        Debug.Assert(FrameCount % (values.Length / 4) == 0, "PropertyItem has invalid value byte array. The FrameCount should be evenly divisible by a quarter of the byte array's length.");
+
+                        _frameEndTimes = new int[FrameCount];
+
+                        for (int f = 0, i = 0; f < FrameCount; ++f, i += 4)
                         {
-                            _frameDelay[i] = values[i * 4] + 256 * values[i * 4 + 1] + 256 * 256 * values[i * 4 + 2] + 256 * 256 * 256 * values[i * 4 + 3];
+                            if (i == values.Length)
+                            {
+                                i = 0;
+                            }
+
+                            // Frame delays are stored in 1/100ths of a second; convert to milliseconds while accumulating
+                            _frameEndTimes[f] = (f > 0 ? _frameEndTimes[f - 1] : 0) + (BitConverter.ToInt32(values, i) * 10);
```

Comment:
>BitConverter.ToInt32 [](http://example.com/codeflow?start=87&length=20)

is it needed or simple cast can do it?

Reply:
The previous code was doing the work of BitConverter manually by multiplying the bytes by the appropriate multiples of 256 and adding the results together. I could revert back to that, but I wasn't sure of a better way to take 4 bytes from the array and convert to a single int. Let me know if you had something in mind that would be simpler.

Reply:
no need to change. I was just wondering. I think BitConverter would be better to handle the big and little Endian cases.